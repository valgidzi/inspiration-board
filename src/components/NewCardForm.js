import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

class NewCardForm extends Component {
  constructor() {
    super();

    this.state = {
      text: "",
      emoji: ""
    }
  }

  onInputChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;
    const newState = {};
    newState[field] = value;
    this.setState(newState);
  }

  onFormSubmit = (event) => {
    event.preventDefault();
    const newCard = {
      text: this.state.text,
      emoji: this.state.emoji
    };
    this.props.addCardCallback(newCard);

    this.setState({
      text: '',
      emoji: ''
    });

  }

  render() {
  const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]
  const dropdownEmojis = EMOJI_LIST.map((emojiText, i) => {
    return <option
      key={i} value={emojiText}>{emoji.getUnicode(emojiText)}</option>
  })
    return (
      <form className="new-card-form__form" id="newcardform"
          onSubmit={this.onFormSubmit}>
          <label htmlFor="Text" className="new-card-form__form-label">Text</label>
          <textarea
            className="new-card-form__form-textarea"
            name="text"
            form="newcardform"
            value={this.state.text}
            onChange={this.onInputChange}/>
          <label htmlFor="emoji" className="new-card-form__form-label">Emoji</label>
          <select className="new-card-form__form-select"
            name="emoji"
            value={this.state.emoji}
            onChange={this.onInputChange}>
            {dropdownEmojis}
          </select>
        <input type="submit" value="Create Card" className="new-card-form__form-button"/>
      </form>
    )
  }
}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func,
};

export default NewCardForm;
