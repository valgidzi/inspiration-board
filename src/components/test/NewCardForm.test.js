import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it matches an existing snapshot', () => {

    const wrapper = shallow( <NewCardForm text="text" emoji="emoji"/>);

    expect(wrapper).toMatchSnapshot();
  });
});
