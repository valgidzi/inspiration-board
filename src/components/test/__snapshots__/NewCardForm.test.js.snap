// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot 1`] = `
<form
  className="new-card-form__form"
  id="newcardform"
  onSubmit={[Function]}
>
  <label
    className="new-card-form__form-label"
    htmlFor="text"
  >
    Text
  </label>
  <textarea
    className="new-card-form__form-textarea"
    form="newcardform"
    name="text"
    onChange={[Function]}
    value=""
  />
  <label
    className="new-card-form__form-label"
    htmlFor="emoji"
  >
    Emoji
  </label>
  <select
    className="new-card-form__form-select"
    name="emoji"
    onChange={[Function]}
    value=""
  >
    <option
      key="0"
      value=""
    />
    <option
      key="1"
      value="heart_eyes"
    >
      😍
    </option>
    <option
      key="2"
      value="beer"
    >
      🍺
    </option>
    <option
      key="3"
      value="clap"
    >
      👏
    </option>
    <option
      key="4"
      value="sparkling_heart"
    >
      💖
    </option>
    <option
      key="5"
      value="heart_eyes_cat"
    >
      😻
    </option>
    <option
      key="6"
      value="dog"
    >
      🐶
    </option>
  </select>
  <input
    className="new-card-form__form-button"
    type="submit"
    value="Create Card"
  />
</form>
`;
